### PR TITLE
Fix broken docs links

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
Fixes the broken links that can be fixed manually.

Its out of scope for this PR to solve the general issue https://github.com/holoviz/panel/issues/7280 with broken links between reference notebooks.

Its also out of scope to clean up false positives like complaints about internal links `#...` not found.

Closes https://github.com/holoviz/panel/issues/7087

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -95,6 +95,7 @@
     'sphinx.ext.mathjax',
     'sphinx.ext.ifconfig',
     'sphinx.ext.linkcode',
+    'sphinx.ext.inheritance_diagram',
```
Comment (MarcSkovMadsen) [inline doc/conf.py:98]: An attempt to fix 358 errors like `/home/jovyan/repos/private/panel/doc/api/panel.__main__.rst:4: ERROR: Unknown directive type "inheritance-diagram".`

**Comment by MarcSkovMadsen** [comment]
I've done a first iteration. Now I need to rebuild docs to see if problems have been fixed without introducing other issues.

File: doc/conf.py [hunk 2]
```diff
@@ -86,6 +86,7 @@
 
 
 extensions = [
+    'bokeh.sphinxext.bokeh_plot',
```
Comment (MarcSkovMadsen) [inline doc/conf.py:89]: This is an attempt to fix 4 errors like `docstring of panel.models.widgets.CheckboxButtonGroup.button_type:4: ERROR: Unknown directive type "bokeh-plot".`.

File: doc/about/releases.md
Comment (MarcSkovMadsen) [inline doc/about/releases.md:1]: The webllm.md file should be added to some toctree similarly to rest of gallery examples. But I could not figure out where. Can you fix @philippjfr ?

/home/jovyan/repos/private/panel/doc/gallery/webllm.md: WARNING: document isn't included in any toctree

Comment (MarcSkovMadsen) [reply doc/about/releases.md:1]: Maybe its related to /home/jovyan/repos/private/panel/doc/gallery/index.md:186: WARNING: 'myst' cross-reference target not found: 'WebLLM' [myst.xref_missing]

**Comment by MarcSkovMadsen** [comment]
I believe this fixes all or most broken links that should be fixed manually.

File: doc/conf.py (hunk 1 above)
Comment (philippjfr) [reply doc/conf.py:99]: Hmm, that should have been disabled, I actually don't like the inheritance diagrams, but I'll re-enable for now.

**Review by philippjfr — APPROVED** [review]

**Pull request merged**